homeassistant/components/osoenergy/water_heater.py

```diff
@@ -46,14 +58,116 @@ async def async_setup_entry(
         return
     async_add_entities((OSOEnergyWaterHeater(osoenergy, dev) for dev in devices), True)
 
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        SERVICE_GET_PROFILE,
+        {},
+        OSOEnergyWaterHeater.async_get_profile.__name__,
+        supports_response=SupportsResponse.ONLY,
+    )
+
+    service_set_profile_schema = cv.make_entity_service_schema(
+        {
+            vol.Optional(f"hour_{hour:02d}"): vol.All(
+                vol.Coerce(int), vol.Range(min=10, max=75)
+            )
+            for hour in range(24)
+        }
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_PROFILE,
+        service_set_profile_schema,
+        OSOEnergyWaterHeater.async_set_profile.__name__,
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_V40MIN,
+        {
+            vol.Required(ATTR_V40MIN): vol.All(
+                vol.Coerce(float), vol.Range(min=200, max=550)
+            ),
+        },
+        OSOEnergyWaterHeater.async_set_v40_min.__name__,
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_TURN_OFF,
+        {vol.Required(ATTR_UNTIL_TEMP_LIMIT): vol.All(cv.boolean)},
+        OSOEnergyWaterHeater.async_oso_turn_off.__name__,
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_TURN_ON,
+        {vol.Required(ATTR_UNTIL_TEMP_LIMIT): vol.All(cv.boolean)},
+        OSOEnergyWaterHeater.async_oso_turn_on.__name__,
+    )
```

Comment:
Is it possible to overwrite the default services? cc @emontnemery

Reply:
The services have the same name but are created in the domain of the integration. They do not override the default water_heater turn_on and turn_off services.